### PR TITLE
Adding count of viewed documents to Reader document list

### PR DESCRIPTION
**Pull request by mogrenAtWork**
- Connects #2498 

Showing the number of documents viewed and the total documents above the accordion on the document list

# Before
screen shot 2017-07-30 at 9 52 03 pm

# After
screen shot 2017-07-30 at 9 45 08 pm

### CODE REVIEW

File: client/app/reader/ClaimsFolderDetails.jsx [hunk 1]
```diff
@@ -34,11 +34,20 @@ class ClaimsFolderDetails extends React.PureComponent {
   }
 
   render() {
-    const { appeal } = this.props;
+    const { appeal, documents } = this.props;
     const appealDoesntExist = _.isEmpty(appeal);
```
Comment (mogrenAtWork) [inline client/app/reader/ClaimsFolderDetails.jsx]: Do we have cases where this component would be used without an appeal?

File: client/app/reader/ClaimsFolderDetails.jsx [hunk 2]
```diff
@@ -34,11 +34,20 @@ class ClaimsFolderDetails extends React.PureComponent {
...
+    const docsViewedCount = documents.reduce(
```
Comment (NickHeiner) [inline client/app/reader/ClaimsFolderDetails.jsx]: This should be:

```js
_.filter(documents, 'opened_by_current_user').length
```

**Comment by NickHeiner** [comment]
LGTM aside from my one comment. I did not pull this down and run it locally.

File: client/app/reader/ClaimsFolderDetails.jsx (hunk 1 above)
Comment (tejans24) [inline client/app/reader/ClaimsFolderDetails.jsx]: When the appeal being fetched from the API, a loading message is shown. Thus, the component is shown without an appeal existing in the store.

File: client/app/reader/ClaimsFolderDetails.jsx [hunk 3]
```diff
@@ -34,11 +34,20 @@ class ClaimsFolderDetails extends React.PureComponent {
...
+      (sum, doc) => {
+        return doc.opened_by_current_user ? sum + 1 : sum;
+      }, 0);
 
     return <div className="cf-claims-folder-details">
-      {!appealDoesntExist && <h1>{appeal.veteran_full_name}'s Claims Folder</h1>}
+      {!appealDoesntExist && <div>
+        <h1 className="cf-push-left">{appeal.veteran_full_name}'s Claims Folder</h1>
```
Comment (tejans24) [inline client/app/reader/ClaimsFolderDetails.jsx]: The document view count message won't be shown if the appeal API call fails or takes a while to load. That's indicated by appealDoesntExist. I think the doc view count should be independent from the appeal being loaded.

**Comment by tejans24** [comment]
LGTM! 👍✅

**Pull request merged**